### DISCUSSION TITLE
On event send switchToProfile, Smart Profile stop working

### DISCUSSION THREAD

**Discussion by kekeljevic** [elgatosf/elgato#117] [Help] [answered]
Hi, I'm experiencing the exact same problem as described here: [https://github.com/orgs/elgatosf/discussions/5](url)

My predefined profile for plugin works fine, with smart profile switching, until I use event sent switchToProfile to go to another page. It switch to another page fine, but after it smart profiles stop working. 
Only option to get them working again is to restart Streamdeck app or to manually switch back to default profile. 

Help ... 

My setup:
  Device: Stream Deck
  Software Version: 6.6.0 (20583)
  Firmware Version: 1.02.000
  OS: Windows 11

My code:
  var json = {
	  "event": "switchToProfile",
	  "context": SD_pluginID,
	  "device": SD_deviceID,
	  "payload": {
		  "profile": "My Plugin Profile",
		  "page": 1
	  }
  }
  SD_Socket.send(JSON.stringify(json));

**Comment by ChekTek** [comment, marked as answer]
Hi @kekeljevic 

I just tested this out on Stream Deck version `6.6.0 (20583)` and it appears to be working as expected. When a plugin calls `switchToProfile` with a predefined profile, it will prioritize this profile over any smart profiles. If you want to re-enable smart profiles, the plugin can call `switchToProfile` without a profile parameter and Stream Deck will return to the profile that it was previously on, and now smart profiles will take priority again. 

We are actively working on something we are calling "Full Canvas Control" which will allow for a single action to take over a full Stream Deck canvas. This may make it easier to manage these types of situations in the future.

**Comment by kekeljevic** [comment]
> Hi @kekeljevic
> 
> I just tested this out on Stream Deck version `6.6.0 (20583)` and it appears to be working as expected. When a plugin calls `switchToProfile` with a predefined profile, it will prioritize this profile over any smart profiles. If you want to re-enable smart profiles, the plugin can call `switchToProfile` without a profile parameter and Stream Deck will return to the profile that it was previously on, and now smart profiles will take priority again.
> 
> We are actively working on something we are calling "Full Canvas Control" which will allow for a single action to take over a full Stream Deck canvas. This may make it easier to manage these types of situations in the future.

In case someone encounters a similar issue, I calls switchToPage with an empty payload whenever the connection to the application is closed. Thank You so much for pointing me in the right direction.
